### ISSUE TITLE
511 is now "stable" and the viewport freezing bug still exists

### ISSUE THREAD

**Issue by BobdaBiscuit** [d3athrow/vgstation13#14685]
This is a duplicate I know but it's pretty urgent as it's a huge issue affecting many players and from what I can see nearly no work has been done to fix it.

What we know so far is it's brought on by 511 and fixed by downgrading to 510, the viewport stops updating and becomes a frozen still image.
This isn't just the game loading/lag as animations on static objects  such as airvents and taser shots freeze completely.
Chat/sound etc all works as normal.

I assume it isn't linked to any specific windows version as I get it on linux (windows xp and 7) along with just how widespread it is.

I'm too shite at coding to know how the backend of the viewport works but does anyone have any idea on how we can fix this?

https://github.com/vgstation-coders/vgstation13/issues/13241

**Comment by BobdaBiscuit** [comment]
Average is around 2 people per round can't play due to this.

**Comment by RealestEstate** [comment]
I got this problem and was told to report it with as much info I can provide. after playing a round with no problems, I reconnected to find the text box updating the data for the new round, but I didn't have a lobby tab. I reconnected, and got a black screen instead of the lobby splash screen, but the lobby tab was working. once the round started, it was just black on the screen, but the text box was updating. I reconnected again, all the times so far using the reconnect command, and the screen updated and froze again, so I could see me sitting in the chair, but it was a still image (vents didn't have their animation, moving unresponsive on the screen, but I could hear me crashing into airlocks.) then I tried shutting off byond, and turning it back on and logging back in, still no luck. eventually I switched over to stable and it worked.

tldr; I might have been playing on an outdated version of Byond like a total mong without realizing it, but I was told to report it so here I am.

**Comment by BobdaBiscuit** [comment]
From my experience with the bug it never occurs in my first login for a round, if I disconnect and reconnect after that there's a chance the bug will occur and won't go away until the server restarts.

When reconnecting the viewport works as normal and is responsive for around 0.5 - 2 seconds before freezing completely.
Attempting to resize the viewport turns it completely black.
Attempting to enter text mode displays the text, freezes again and refuses to go back to normal when turned off.

This is with Arch Linux, Wine Staging in Windows XP mode.

**Comment by SonixApache** [comment]
I've had this shit happen to me mostly as AI. And I feel it takes longer to 'recover from' the longer a round progresses.
There's been times in 3+ hour rounds where I couldn't get the viewport to re-initialize at all.

**Comment by BobdaBiscuit** [comment]
From the discord:

Alright so got another batch of the station screen freezes, black on reconnect, re-freeze on relog, cache clearing generally doesn't work but it'll be fine after a new round starts until it happens again. Version was byond stable 511,1380. 1 about 20 minutes ago in cargo bay just after digging stuff out of a junk pile. There wasn't an abnormal amount of stuff both on screen and it wasn't a large pile. Got another last night with like 8 other people by deff botany but I think that was more to do with the chicken/monkey resprites than anything. I've had freezes in a normal deff atmos, meta tool storage, many other places and times. At first I thought it was a right-click issue but it hasn't been consistent in any location/recent action or reproduce-able. Sorry that this isn't helpful.

**Comment by PJB3005** [comment]
Just contact lummox, ask for a way to get debug info for him and then play until you can reproduce it. That's the best way you'd solve this.

**Comment by jknpj** [comment]
fixed on 512
